application/comit_node/src/bam_api/rfc003/swap.rs

```diff
@@ -36,47 +30,59 @@ pub fn swap_config(
         ],
         move |request: Request| {
             let swap_protocol = header!(request.get_header("swap_protocol"));
-
             match SwapProtocols::from_bam_header(swap_protocol).unwrap() {
                 SwapProtocols::Rfc003 => {
                     let swap_id = SwapId::default();
-                    let (response_sender, response_receiver) = oneshot::channel();
 
-                    if let Ok(swap_request) = decode_request(&request) {
-                        let request_kind =
-                            rfc003::bob::SwapRequestKind::BitcoinEthereumBitcoinQuantityEtherQuantity(
-                                swap_request,
-                            );
-                        sender.unbounded_send((swap_id, request_kind, response_sender)).unwrap();
+                    if let Ok(swap_request) =
```

Comment:
IMO We need to discuss how to remove this `ledgers^2` repetition of things rather than hide it in functions (here is not the only problem). I have some ideas so I would leave it like this for now.